src/app/recover-funds/[address]/page.tsx

```diff
@@ -1,53 +1,78 @@
 'use client';
-import { getTargetChainId } from '@/utils';
-import { utils } from 'ethers';
+import { mapChainIdToName, supportedL2Networks } from '@/utils/network';
+import { Address } from '@arbitrum/sdk';
+import { constants, utils } from 'ethers';
 import dynamic from 'next/dynamic';
 import React, { useEffect, useState } from 'react';
 import { useNetwork } from 'wagmi';
-import {
-  getData,
-  hasBalanceOverThreshold,
-  OperationInfo,
-} from './RecoverFunds';
+import { hasBalanceOverThreshold, OperationInfo } from './RecoverFunds';
 import { RecoverFundsButton } from './RecoverFundsButton';
+import { JsonRpcProvider } from '@ethersproject/providers';
 
 const RecoverFunds = dynamic(() => import('./RecoverFunds'), {
   ssr: false,
 });
 
-const RecoverFundsPage = ({
-  params: { address },
-}: {
-  params: { address: string };
-}) => {
-  const { chain } = useNetwork();
-  const [operationInfo, setOperationInfo] = useState<OperationInfo | null>(
-    null,
-  );
-  const [destinationAddress, setDestinationAddress] = useState<string | null>(
-    null,
-  );
+type OperationInfoByChainId = {
+  [chainId: string]: OperationInfo;
+};
+async function getOperationInfoByChainId(
+  address: string,
+): Promise<OperationInfoByChainId> {
+  // First, obtain the aliased address of the signer
+  const destinationAddress = new Address(address);
+  const { value: aliasedAddress } = destinationAddress.applyAlias();
 
-  const targetChainID = getTargetChainId(chain?.id);
+  // And get its balance to find out the amount we are transferring
+  const operationInfoPromises = Object.entries(supportedL2Networks).map(
+    async ([chainId, rpcURL]) => {
+      const l2Provider = new JsonRpcProvider(rpcURL);
 
-  useEffect(() => {
-    if (!targetChainID) {
-      return;
-    }
+      try {
+        const aliasedSignerBalance = await l2Provider.getBalance(
+          aliasedAddress,
+        );
 
-    getData(targetChainID, address).then((data) => {
-      setOperationInfo(data);
-    });
-  }, [address, targetChainID]);
+        return {
+          balanceToRecover: hasBalanceOverThreshold(aliasedSignerBalance)
+            ? aliasedSignerBalance
+            : constants.Zero,
+          aliasedAddress,
+          chainId,
+        };
+      } catch (e) {
```

Comment:
If the RPC fails for whatever reason, the page will tell the user that there are no funds to recover, but that might be misleading.
Is there any other reason why the flow would catch an error here?
If a faulty RPC is the only reason, I would maybe log/show an internal error somewhere, instead of returning a 0 balance.